### PR TITLE
Bump aeson and cardano node

### PR DESCRIPTION
**Pull request by ghost**
I have updated the code to aeson >= 2 together with cardano-node dependency. I had to put some stubs because `BabbageEra` type is not exported by `Cardano.Api` yet but it compiles and is enough for `plutus-apps`.

Later I understood that it would be nice to separate aeson/cardano-node modifications but it was too late. Hope that this PR will be useful for you to some extent.

### CODE REVIEW

File: lib/core/src/Cardano/Wallet/DB/Sqlite/CheckpointsOld.hs [hunk 1]
```diff
@@ -718,68 +704,3 @@ selectRndStatePending wid = do
   where
     assocFromEntity (RndStatePendingAddress _ accIx addrIx addr) =
         ((W.Index accIx, W.Index addrIx), addr)
-
```
Comment (sevanspowell) [inline lib/core/src/Cardano/Wallet/DB/Sqlite/CheckpointsOld.hs:721]: Reviewer's note, ouroborus-network provides it's own ReaderT instance for MonadSTM (https://github.com/input-output-hk/ouroboros-network/blob/ad47441e9e399495579d7dd05ccaf96b46a487b2/io-classes/src/Control/Monad/Class/MonadSTM.hs#L897)

**Comment by sevanspowell** [comment]
Thank you for your work on this @ak3n. I will take a look now <3

**Comment by sevanspowell** [comment]
bors try

**Comment by Anviking** [comment]
Thanks @ak3n, this was very helpful! I have cherry-picked this into https://github.com/input-output-hk/cardano-wallet/pull/3280

**Pull request closed without merging**